.github/workflows/publish.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish
+on:
+  workflow_run:
+    workflows: [CI]
+    branches: [master]
+    types: [completed]
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  publish:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 7
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20.x
+          cache: "pnpm"
+
+      - run: pnpm install --frozen-lockfile
+      - name: Create Release Pull Request or Publish
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          publish: pnpm run release
```

Comment:
I don't know about `pnpm`, but with `npm` one needs `--access public` and we also recommend `--provenance`.

More about [provenance](https://docs.npmjs.com/generating-provenance-statements).

There you have examples on how to deploy from a Github action.

Or take a look at the publish workflow in a [recent package we published](https://github.com/dfinity/verifiable-credentials-sdk/blob/main/.github/workflows/publish.yml).